apisix/plugins/redirect.lua

```diff
@@ -157,7 +157,7 @@ function _M.rewrite(conf, ctx)
     if conf.http_to_https and _scheme == "http" then
         -- TODO： add test case
         -- PR: https://github.com/apache/apisix/pull/1958
-        if ret_port == 443 then
+        if ret_port == 443 or ret_port == 0 then
```

Comment:
Why check if it's equal to `0`? may be `if not ret_port or ret_port == 443`?

Reply:
`0` can pass through `x-forwarded-port`

checking condition `ret_port == nil or ret_port == 443 or ret_port <= 0 or ret_port > 65535`

Reply:
Gotcha.